test.js

```diff
@@ -0,0 +1,124 @@
+const app = require('express')()
+const bodyParser = require('body-parser')
+const multer = require('multer') // v1.0.5
+const upload = multer() // for parsing multipart/form-data
+
+app.use(bodyParser.json()) // for parsing application/json
+app.use(bodyParser.urlencoded({ extended: true })) // for parsing application/x-www-form-urlencoded
+
+app.post('/profile', upload.array(), function (req, res, next) {
+  console.log(req.body)
+  res.json(req.body)
+})
+
+const port = 3001;
+
+app.get('/', (req, res) => res.send('Hello World!'));
+
+
+class Treenode {
+
+  constructor(label) {
+    this.id = (++Treenode.id_gen).toString()
+    // Treenode.lineage[this.id]=[label]
+    this.label =  label
+    this.parent_id = null
+    // this.parent_label = null
+    this.children = []
+    // Treenode.label_id[this.id] = label
+  }
+
+  add_child(child){
+    this.children.push(child)
+    child.parent_id = this.id
+  }
+
+}
+
+function find_id(iter, parent_id) {
```

Comment:
something like this I would put in the theoretical `TreeNodeManager` 

right now its a problem because in terms of organization its just hanging around without any sense of which module it belongs to

Reply:
argument naming of `iter` and `parent_id` is a little strange, did you mean `node` and `queryId`?